### PR TITLE
#160618102: Enable social login

### PR DESCRIPTION
**Pull request by Phunmbi**
### What does this PR do?
This pull request makes it possible for users to sign in with social accounts like Facebook, Google and Twitter.

#### Description of Task to be completed?
- Have the following endpoints working
 `/api/auth/google`
 `/api/auth/twitter`
 `/api/auth/facebook`
- Authenticate users that sign up/ sign in via this medium.
- Update Swagger documentation with new routes.
- Set up tests for the social login authentication.

#### How should this be manually tested?
- Setup the repo
- run `npm install`
- if a local dev and test database already exist, delete the models and run `sequelize db: migrate` and `sequelize db:migrate --env test`. This is to effect the change in the users model.
- run `npm run test`

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#160618102](https://www.pivotaltracker.com/story/show/160618102)

#### Screenshots (if appropriate)
N/A

#### Questions:
None

### CODE REVIEW

File: lib/mockStrategy.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ * Class representing a strategy
+ * @extends passport.Strategy
+ */
+class Strategy extends passport.Strategy {
+/**
+ * @constructor
+ * @param {string} name - the name of the strategy
+ * @param {function} strategyCallback - the function to handle the callback from the strategy
+ */
+  constructor(name, strategyCallback) {
```
Comment (tonyguesswho) [inline lib/mockStrategy.js:19]: I think a switch statement will make more sense

Comment (Phunmbi) [reply lib/mockStrategy.js:19]: I've implemented this.

Comment (tonyguesswho) [reply lib/mockStrategy.js:19]: ok looks better now

File: lib/mockStrategy.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  */
+  authenticate(req) {
+    if (req.body.access_token === this.user.token) {
+      count += 1;
+
+      if (count === 3 || count === 7 || count === 11) {
+        this.user.id = 'randomwrongid';
+      }
+
+      if (count === 4 || count === 8 || count === 12) {
```
Comment (tonyguesswho) [inline lib/mockStrategy.js]: I still see some nested if statements

**Review by tonyguesswho — APPROVED** [review]
I see all my recommendations have been implemented, this PR can be merged now

File: lib/mockStrategy.js (hunk 2 above)
Comment (Phunmbi) [reply lib/mockStrategy.js]: Resolved

**Review by damafeez — COMMENTED** [review]
LGTM (Looks Good To Me)

**Review by damafeez — COMMENTED** [review]
LGTM (Looks Good To Me)

**Review by damafeez — COMMENTED** [review]
LGTM (Looks Good To Me)

**Review by VictorUkafor — COMMENTED** [review]
This looks cool to me, I love the comments and references in your code. You just have to fix the merge conflict

**Pull request merged**